src/librustc/ty/constness.rs

```diff
@@ -68,16 +69,53 @@ impl<'tcx> TyCtxt<'tcx, 'tcx> {
 
 
 pub fn provide<'tcx>(providers: &mut Providers<'tcx>) {
-    /// only checks whether the function has a `const` modifier
+    fn is_intrinsic_promotable(tcx: TyCtxt<'tcx, 'tcx>, def_id: DefId) -> bool {
```

Comment:
It's not about promotable, even though the code used to suggest that. It's about const evaluability